client/src/pages/MySitters/MySitters.tsx

```diff
@@ -1,33 +1,182 @@
 /* eslint-disable prettier/prettier */
 import { useState, useEffect } from "react";
 import getRequests from '../../helpers/APICalls/getRequests';
+import updateRequest from '../../helpers/APICalls/updateRequest';
 import { RequestData } from "../../interface/Request";
-import { Container, Grid, Box } from '@material-ui/core';
+import { 
+  Container, 
+  Grid, 
+  Box, 
+  Card, 
+  CardHeader, 
+  CardContent, 
+  IconButton,
+  Typography 
+} from '@material-ui/core';
+import { Settings } from "@material-ui/icons";
 import useStyles from "./useStyles";
+import BookingCard from "../../components/BookingCards/BookingCard";
+import BookingsDatePicker from "../../components/BookingsDatePicker/BookingsDatePicker";
+
+const mockImg = 'https://badairies.co.uk/assets/admin/plugins/images/users/4.jpg';
 
 export default function MySitters(): JSX.Element {
   const classes = useStyles();
 
   const [requests, setRequests] = useState<RequestData[]>([]);
+  const [dates, setDates] = useState<Date[]>([])
+  const [cardAction, setCardAction] = useState<boolean>(false);
+  const [selectedBooking, setSelectedBooking] = useState<string>("");
 
   useEffect(() => {
     getRequests()
-      .then(data => setRequests(data));
+      .then(data => {
+        setRequests(data)
+        const dates: Date[] = [];
+        data.map(booking => {
+          if (booking.accepted) {
+            const bookingDate = new Date(booking.start)
+            dates.push(bookingDate)
+          }
+        });
+        setDates(dates);
+    });
```

Comment:
We chould do async/await here. The structure for doing asynchronous calls in our effect hooks is as follows.

```js
useEffect(() => {
  const getRequests = async () => { /* fetch here */ };
  getRequests() // call the function in the useEffect.
}, [])
```

Reply:
Solved